### PR TITLE
web: remove unimported modules from the `web` workspace

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

While working on the Webpack upgrade, I noticed multiple unused modules in the `web` package. This PR removes unimported files from the codebase. 🧹

### CODE REVIEW

File: client/web/src/insights/hooks/use-insight-dashboard.ts [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-import { useDashboards } from './use-dashboards/use-dashboards'
-
-export interface UseInsightDashboardsProps extends SettingsCascadeProps<Settings> {
-    insightId: string
-}
-
-/**
- * Returns all dashboards (built-in - personal or org) that have insight Id
- */
-export function useInsightDashboards(props: UseInsightDashboardsProps): RealInsightDashboard[] {
```
Comment (valerybugakov) [inline client/web/src/insights/hooks/use-insight-dashboard.ts:17]: @vovakulikov, is it required for an upcoming feature?

Comment (vovakulikov) [reply client/web/src/insights/hooks/use-insight-dashboard.ts:17]: Hey @valerybugakov. I think was introduced for the first time for the dashboard feature. But we didn't have enough time to finish that. I think this hook was intended for this https://github.com/sourcegraph/sourcegraph/issues/22922

We still have it in our roadmap so I would leave this hook for now (but you're right at the moment we don't have any consumers for this `useInsightDashboards`)

**Review by eseliger — APPROVED** [review]

File: client/web/src/site/PerformanceWarningAlert.tsx [hunk 2]
```diff
@@ -1,34 +0,0 @@
...
-import { eventLogger } from '../tracking/eventLogger'
-
-const onClickCTA = (): void => {
-    eventLogger.log('AlertPerformanceWarningCTAClicked')
-}
-
-/**
- * An alert that explains the performance limitations of single-node Docker deployments.
- */
-export const PerformanceWarningAlert: React.FunctionComponent = () => (
```
[the remark this answers is not in the document]
Comment (limitedmage) [reply client/web/src/site/PerformanceWarningAlert.tsx:13]: This was not added to the new streaming search results UI... I wonder if we need this? I don't think the streaming search reports the need for this? @sourcegraph/search-product  @sourcegraph/search-core  any ideas here?

[the remark this answers is not in the document]
Comment (limitedmage) [reply client/web/src/site/PerformanceWarningAlert.tsx:13]: (I think it's OK to delete this since it's not used. If we need to, we can add it back)

**Review by limitedmage — APPROVED** [review]

**Pull request merged**